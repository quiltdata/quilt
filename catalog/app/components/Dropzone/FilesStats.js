import * as React from 'react'
import * as M from '@material-ui/core'

import { readableBytes } from 'utils/string'

const useStyles = M.makeStyles(() => ({
  icon: {
    marginLeft: 4,
  },
}))

export default function FilesStats({ files, warning }) {
  const classes = useStyles()

  const totalSize = React.useMemo(() => files.reduce((sum, f) => sum + f.size, 0), [
    files,
  ])

  if (!files.length) return null

  return (
    <span>
      : {files.length} ({readableBytes(totalSize)})
      {warning && <M.Icon className={classes.icon}>error_outline</M.Icon>}
    </span>
  )
}
