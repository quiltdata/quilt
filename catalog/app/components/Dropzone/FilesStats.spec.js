import * as React from 'react'
import renderer from 'react-test-renderer'

import FilesStats from './FilesStats'

describe('components/Dropzone/FilesStats', () => {
  it('should render', () => {
    const tree = renderer.create(<FilesStats files={[]} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  // FIXME: fix "jest + react-intl"
  it.skip('should render files total size', () => {
    const tree = renderer
      .create(<FilesStats files={[{ size: 100000 }, { size: 200000 }]} />)
      .toJSON()
    expect(tree).toMatchInlineSnapshot()
  })
})
