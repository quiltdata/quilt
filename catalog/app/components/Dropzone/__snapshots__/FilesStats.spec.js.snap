// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Dropzone/FilesStats should render 1`] = `null`;
